app/controllers/teams_controller.rb

```diff
@@ -68,6 +68,15 @@ def edit
     @team = Team.find(params[:id])
   end
 
+  def delete_all
+    # Deletes all the teams assigned to a course, by finding them via node_object_id
+    root_node = Object.const_get(session[:team_type] + "Node").find_by(node_object_id: params[:id])
+    child_nodes = root_node.get_teams.map { |e| e.node_object_id}
+    course_teams = Team.where(id: child_nodes)
+    course_teams.destroy_all if course_teams
```

Comment:
I have one small suggestion to make on lines 75-76. Instead of fetching the Team data and then deleting, we can directly delete the data from child_nodes array. I was able to make the following code run achieving same results:
```ruby
Team.delete_all(id:child_nodes) if child_nodes
```
It has one less db call.

Reply:
I think this actually won't be saving any DB calls, as :-
1. Rails does a lazy evaluation of database queries so it won't be evaluated until we call destroy
2. Also, rails cache the query result so if there is the same DB call, it will be using the cached result instead of fetching it again from the database.

If you want to test the number of queries made, you can install a gem called 'db-query-matchers' on your local branch and test the number of queries made in each case. 
Let me know if you think I missed something :)

Reply:
Sure! I will install the gem. If it looks good to you, I'll merge it.